src/player.cpp

```diff
@@ -4151,7 +4151,7 @@ void player::apply_damage(Creature *source, body_part hurt, int dam)
         hp_cur[hurtpart] = 0;
     }
 
-    if( hp_cur[hurtpart] <= 0 ) {
+    if( source->is_hallucination() != true && hp_cur[hurtpart] <= 0 ) {
```

Comment:
Why `foo != true` instead of just `!foo`? You must also check `source` being non-null as there are several calls to that function that supply `nullptr` as first argument.